### PR TITLE
Rework pv check to check StorageClasses and CSIDrivers

### PR DESCRIPTION
**Pull request by jsafrane**
Rework pv check to check StorageClasses and CSIDrivers too.
* Rename top storage-obj-save / storage-obj-check
* Mark it as `best_effort` not to fail the whole CI job when some leftover objects are found.

cc @openshift/storage

### CODE REVIEW

File: ci-operator/step-registry/openshift/e2e/vsphere/csi/openshift-e2e-vsphere-csi-workflow.yaml [hunk 1]
```diff
@@ -1,18 +1,19 @@
...
     - ref: storage-conf-wait-for-csi-driver
     # Wait for CSI migration enabed by TechPreviewNoUpgrade to restart all nodes
     - ref: storage-conf-wait-for-csi-migration
-    - ref: storage-pv-save
+    - ref: storage-obj-save
     test:
     - ref: openshift-e2e-test
     post:
-    - ref: storage-pv-check
+    - ref: storage-obj-check
```
Comment (gnufied) [inline ci-operator/step-registry/openshift/e2e/vsphere/csi/openshift-e2e-vsphere-csi-workflow.yaml:16]: can we stop this from running if starting the cluster failed? It seems pointless to check for storage objects if cluster failed to come up. It causes some noise in CI.

Comment (jsafrane) [reply ci-operator/step-registry/openshift/e2e/vsphere/csi/openshift-e2e-vsphere-csi-workflow.yaml:16]: This PR makes the `post` steps optional (i.e. do not complain if it fails). It's not possible to skip `pre`  steps though.

I updated `storage-obj-save` to never fail, see the script.

File: ci-operator/step-registry/storage/obj/check/storage-obj-check-ref.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  timeout: 10m
+  as: storage-obj-check
+  from: tests
+  commands: storage-obj-check-commands.sh
+  resources:
+    requests:
+      cpu: 10m
+      memory: 100Mi
+  documentation: |-
+    This step compares existing stroage objects (PVs, StorageClasses, CSIDrivers)
```
Comment (bertinatto) [inline ci-operator/step-registry/storage/obj/check/storage-obj-check-ref.yaml:13]: nit: storage

**Review by bertinatto — COMMENTED** [review]
/lgtm

File: ci-operator/step-registry/storage/obj/check/storage-obj-check-ref.yaml (hunk 2 above)
Comment (jsafrane) [reply ci-operator/step-registry/storage/obj/check/storage-obj-check-ref.yaml]: Fixed

**Comment by jsafrane** [comment]
/retest
I want to see CSI migration tests...

**Comment by jsafrane** [comment]
Finally, this CI run shows that leaked PV is now a flake:

https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/18555/rehearse-18555-periodic-ci-openshift-release-master-nightly-4.9-e2e-aws-csi-migration/1409432709075308544

`[sig-storage] Ensure that all storage objects are deleted during the tests expand_more` is now under "4/2781 Tests Flaky." And with "open stdout" you get list of extra objects.

**Comment by jsafrane** [comment]
Fixed newly added jobs to use `storage-obj-save` instead of `storage-pv-save`.

File: ci-operator/step-registry/openshift/e2e/gcp/csi/openshift-e2e-gcp-csi-workflow.yaml [hunk 3]
```diff
@@ -1,14 +1,15 @@
 workflow:
   as: openshift-e2e-gcp-csi
   steps:
+    allow_best_effort_post_steps: true
```
Comment (gnufied) [inline ci-operator/step-registry/openshift/e2e/gcp/csi/openshift-e2e-gcp-csi-workflow.yaml:4]: does this have any side effect on other steps/chains being executed in post step?

File: ci-operator/step-registry/storage/obj/save/storage-obj-save-commands.sh [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    $2
+    </system-out>
+  </testcase>
+  <testcase name="$0" time="0"/>
+</testsuite>
+EOF
+    # For build-log.txt
+    echo $MESSAGE
+    echo $STDOUT
+}
```
Comment (gnufied) [inline ci-operator/step-registry/storage/obj/save/storage-obj-save-commands.sh:32]: who is calling this function?

Comment (jsafrane) [reply ci-operator/step-registry/storage/obj/save/storage-obj-save-commands.sh]: Good catch, removed.

**Comment by jsafrane** [comment]
/retest
I can see some flakes. With Kubernetes rebase in origin, I don't see any leaked volumes...

**Comment by jsafrane** [comment]
I can see could of leaked PVs marked as flakes

**Pull request merged**